linode/manager
# M3-5814: Refactors in SelectPlanPanel.tsx / DatabaseCreate.tsx

## Description
Moves the logic for calculating the prices for different DB cluster plans back to the `DatabaseCreate` component. In addition this PR establishes a new `PlanSelectionType` to help standardized the data used to select form different plans. I think more can be done with this but the current iteration is a good first step. I think due to the widespread use of the `SelectPlanPanel` it could be extracted from the `LinodeCreate` structure and generalized to make it easier to possibly create more custom plan selections in the future for new products.

## How to test

Ensure that creation flows for Linodes and Databases work as intended paying attention to pricing and copy. Also ensure that the Linode Resize flow has stayed the same paying close attention to pricing and copy.
